rust/osm2lanes/src/overpass.rs

```diff
@@ -112,7 +119,54 @@ pub async fn get_way(id: ElementId) -> Result<(Tags, Locale), reqwest::Error> {
         elements.pop().unwrap()
     };
     assert_eq!(way_element.r#type, ElementType::Way);
-    assert_eq!(way_element.id, id);
+    assert_eq!(&way_element.id, id);
 
     Ok((way_element.tags, locale))
 }
+
+/// Get Tags and Geometries from Overpass.
+/// Given a longitude and latitude, find a way within 10m.
+/// If there is more than one way within 10m, returns one at random...
+///
+/// # Errors
+///
+/// May occur when processing a request.
+///
+/// # Panics
+///
+/// Unexpected data from overpass and/or openstreetmap.
+///
+pub async fn get_nearby(
+    lng_lat: (f64, f64),
+) -> Result<(Tags, Locale, Vec<LatLon>), reqwest::Error> {
+    const RADIUS: f64 = 10.0_f64;
+    let mut resp = reqwest::Client::new()
+        .get(format!(
+            r#"https://overpass-api.de/api/interpreter?data=[out:json][timeout:25];
+            way
+            (around:{RADIUS},{},{})
+            ["highway"];
+            out tags geom;"#,
+            lng_lat.0, lng_lat.1
+        ))
+        .send()
+        .await?
+        .json::<OverpassResult>()
+        .await?;
+    log::debug!("{:#?}", resp);
+    if resp.elements.len() > 1 {
+        log::warn!("more than one nearby way found, returning one at random");
+    }
+    resp.elements.truncate(1);
+    let element = resp.elements.pop().unwrap();
+    assert_eq!(element.r#type, ElementType::Way);
+    // TODO: single overpass query
```

Comment:
it will probably take me a while to figure out what the query should look like, so now I do it in 2 stages